tests/test_models/test_nn/test_timesfm.py

```diff
@@ -0,0 +1,245 @@
+
+
+@pytest.mark.smoke
+def test_fit(example_tsds):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch")
+    model.fit(example_tsds)
+
+
+@pytest.mark.smoke
+def test_predict(example_tsds):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch")
+    with pytest.raises(NotImplementedError, match="Method predict isn't currently implemented!"):
+        model.predict(ts=example_tsds, prediction_size=1)
+
+
+def test_forecast_warns_big_context_size(ts_increasing_integers):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=512)
+    pipeline = Pipeline(model=model, horizon=1)
+    pipeline.fit(ts_increasing_integers)
+    with pytest.warns(UserWarning, match="Actual length of a dataset is less that context size."):
+        _ = pipeline.forecast()
+
+
+@pytest.mark.parametrize("encoder_length", [32, 64, 128])
+@pytest.mark.parametrize("ts", ["ts_increasing_integers", "ts_nan_start"])
+def test_forecast(ts, expected_ts_increasing_integers, encoder_length, request):
+    ts = request.getfixturevalue(ts)
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=encoder_length)
+    pipeline = Pipeline(model=model, horizon=2)
+    pipeline.fit(ts)
+    forecast = pipeline.forecast()
+    assert_frame_equal(forecast.df, expected_ts_increasing_integers.df, atol=1)
+
+
+def test_forecast_failed_nan_middle_target(ts_nan_middle):
+    model = TimesFMModel(path_or_url="google/timesfm-1.0-200m-pytorch", encoder_length=128)
+    pipeline = Pipeline(model=model, horizon=2)
+    pipeline.fit(ts_nan_middle)
+    with pytest.raises(ValueError, match="There are NaNs in the middle or end of the time series."):
+        _ = pipeline.forecast()
+
+
+@pytest.mark.parametrize("encoder_length", [32, 64, 128])
+@pytest.mark.parametrize("ts", ["ts_increasing_integers", "ts_nan_start"])
+def test_forecast_exogenous_features(ts, expected_ts_increasing_integers, encoder_length, request):
+    ts = request.getfixturevalue(ts)
+
+    horizon = 2
+    transforms = [
+        SegmentEncoderTransform(),
+        LagTransform(in_column="target", lags=[horizon, horizon + 1], out_column="lag"),
+        DateFlagsTransform(day_number_in_week=True, day_number_in_month=False, is_weekend=False, out_column="flag"),
+    ]
+    model = TimesFMModel(
+        path_or_url="google/timesfm-1.0-200m-pytorch",
+        encoder_length=encoder_length,
+        static_categoricals=["segment_code"],
+        time_varying_reals=[f"lag_{horizon}", f"lag_{horizon+1}"],
+        time_varying_categoricals=["flag_day_number_in_week"],
+    )
+    pipeline = Pipeline(model=model, transforms=transforms, horizon=horizon)
+    pipeline.fit(ts)
+    forecast = pipeline.forecast()
+    assert_frame_equal(forecast.df.loc[:, pd.IndexSlice[:, "target"]], expected_ts_increasing_integers.df, atol=1)
+
+
+def test_forecast_exog_features_failed_nan_middle_target(ts_nan_middle):
```

Comment:
We didn't test case when there are missing values for each timestamp. This case is somewhat special.

Reply:
I was sure that we do not create TSDataset if there are all NaNs in segment)

Reply:
There shouldn't be all NaNs in segment. We should for each timestamp have at least one segment that have NaN at this timestamp.

Reply:
I think we cover such case in "ts_exog_all_nan" already.